tasks/Module 4/task_1.py

```diff
@@ -0,0 +1,41 @@
+"""Task 1 of Module 4."""
+from pathlib import Path
+import pyinputplus as pyip
+
+WORDS_TO_REPLACE = ["ADJECTIVE", "NOUN", "VERB"]
+
+
+def mad_libs():
+    """
+    Create and read in text files
+    and lets the user add their own text instead of words in constant.
+    """
+    input_file_path = Path("file.txt")
+    output_file_path = Path("output_file.txt")
+    new_text = []
+
+    if not input_file_path.is_file():
+        with open(input_file_path, "w") as input_file:
+            input_file.write("The ADJECTIVE panda walked to the NOUN and then VERB. "
+                             "A nearby NOUN was unaffected by these events.")
+
+    with open(input_file_path, "r") as input_file:
+        with open(output_file_path, "w") as output_file:
+            for line in input_file:
+                for word in line.split():
+                    if word in WORDS_TO_REPLACE:
+                        input_word = pyip.inputStr(f"Please provide {word}:\n")
+                        new_text.append(input_word)
+                    elif word[:len(word) - 1] in WORDS_TO_REPLACE:
```

Comment:
I suppose that it is for case when word is ended with punctuation mark. If you want you can think about splitting input text using some [smart regular expression](https://stackoverflow.com/questions/367155/splitting-a-string-into-words-and-punctuation), so you won't need this condition.

Reply:
https://github.com/musk-ox right now my idea is to introduce new variable `slitted_sentence = re.findall(r"[\w']+|\.", line)` and then iterate over it, but as a result i'll have a string with a space before the dots(at least if i'll try to `output_file.write(" ".join(new_text))` )

if there a way we could simply connect the data? As right now i see it as we replacing one `elif` with a loop to concatenate it all back :)

Reply:
Good point @GeorgiiDermenzhy. To overcome this we can modify the regular expression by adding `|\s` to include also spaces:
```python
slitted_sentence = re.findall(r"[\w']+|\.|\s", line)
```

Reply:
@musk-ox works perfectly as always, thanks :)